### ISSUE TITLE
-ConfigFile should error when target config is invalid

### ISSUE THREAD

**Issue by bc3tech** [NuGet/Home#388]
In most all nuget commands, specifying `-ConfigFile` is supposed to redirect the configuration for that nuget command run against a specifies configuration file.

it doesn't.

My custom config:

``` xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <packageSources>
        <add key="myfeed" value="http://nuget.mycompany.com/nuget"/>
        <add key="nuget" value="https://www.nuget.org/api/v2/"/>
    </packageSources>
    <solution>
        <solution>
    <add key="disableSourceControlIntegration" value="true" />
  </solution>
</configuration>
```

Notice the malformed `<solution>` tags in there.

Now observe a nuget run without `-ConfigFile`

```
C:\Projects\myproject>nuget sources
Registered Sources:

  1.  nuget.org [Enabled]
      https://www.nuget.org/api/v2/

C:\Projects\myproject>
```

and with `-ConfigFile`

```
C:\Projects\myproject>nuget sources -ConfigFile Nuget.Config
Registered Sources:

  1.  https://www.nuget.org/api/v2/ [Enabled]
      https://www.nuget.org/api/v2/

C:\Projects\myproject>
```

It silently picks the %AppData% config file instead of alerting the user to the bad form of the desired file.

**Comment by yishaigalatzer** [comment]
Thanks for a detailed issue report. Please include the version of nuget.exe you are using.

**Comment by bc3tech** [comment]
2.8.5

**Comment by ChrisSfanos** [comment]
Fix in both 3.0 RTM and 2.8.6

**Comment by emgarten** [comment]
This has been fixed. Invalid nuget.config files will now error. Also only the specified nuget.config will be used. 

Please let me know if you are still seeing any of these issues.
